core/src/test/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoEntityProviderSpec.scala

```diff
@@ -793,6 +799,35 @@ class DataRepoEntityProviderSpec extends AsyncFlatSpec with DataRepoEntityProvid
         |GROUP BY `root`.`zoe`, `root`.`bob`;""".stripMargin
   }
 
+  it should "connect to a Data Repo instance via client" in {
+    /* This test uses the data repo client lib to connect to a mock data repo server.
+        It verifies that the client lib version in use by Rawls is functional
+        and has no inherent runtime errors. Anything beyond this simple connectivity test
+        should likely be an integration test talking to a real Data Repo instance.
+     */
+    val jsonHeader = new Header("Content-Type", "application/json")
+    val mockSnapshotId = java.util.UUID.randomUUID()
+    val mockPort = 32123
+
+    val mockServer = startClientAndServer(mockPort)
+    mockServer.when(
+      request()
+        .withMethod("GET")
+        .withPath(s"/api/repository/v1/snapshots/${mockSnapshotId.toString}")
+    ).respond(
+      response()
+        .withHeaders(jsonHeader)
+        .withBody(s"""{"id":"${mockSnapshotId.toString}"}""")
+        .withStatusCode(StatusCodes.OK.intValue)
+    )
+
+    val dataRepoDAO = new HttpDataRepoDAO("mock", s"http://localhost:$mockPort")
+    val snapshotResponse = dataRepoDAO.getSnapshot(mockSnapshotId, userInfo.accessToken)
+    mockServer.stopAsync()
+
+    snapshotResponse.getId shouldBe mockSnapshotId.toString
+  }
+
```

Comment:
This test fails when the `jersey.inject` library is not included in dependencies (see below), but passes when it is included. We really need this test, because the other tests we have that connect to data repo only run nightly in alpha - without this test, PRs passed as green even though we had runtime failures.